### PR TITLE
feat(trends): Add "Decimal places" as a Trends viz option

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

Solves ZEN-8274.

## Changes

Added a new visualization option (frontend-only) to Trends called "Decimal places":

Screenshot 2023-12-14 at 14 27 15

## How did you test this code?

Updated some frontend tests. Let me know if there are any other tests I should add.

### CODE REVIEW

**Review by thmsobrmlr — COMMENTED** [review]
Super cool and works really well. There's a backend side `filters_to_query` method, where the filter needs to be added as well.

File: frontend/src/queries/nodes/InsightViz/InsightDisplayConfig.tsx [hunk 1]
```diff
@@ -173,3 +184,34 @@ export function InsightDisplayConfig(): JSX.Element {
...
+
+function DecimalPrecisionInput(): JSX.Element {
+    const { insightProps } = useValues(insightLogic)
+    const { trendsFilter } = useValues(insightVizDataLogic(insightProps))
+    const { updateInsightFilter } = useActions(insightVizDataLogic(insightProps))
+
+    const reportChange = useDebouncedCallback(() => {
+        posthog.capture('decimal places changed', {
+            decimal_places: trendsFilter?.decimal_places,
+        })
```
[the remark this answers is not in the document]
Comment (Twixes) [reply frontend/src/queries/nodes/InsightViz/InsightDisplayConfig.tsx:200]: Yeah, I agree with Ben. I don't think `eventUsageLogic` is either helpful nor unhelpful… it's just there as boilerplate, but then I don't see a point in using it. It's rare that we want to capture the same event at two call sites, and for reference purposes searching `posthog.capture(` is just as easy as opening `eventUsageLogic.ts`.

File: frontend/src/queries/nodes/InsightQuery/utils/filtersToQueryNode.ts [hunk 2]
```diff
@@ -254,6 +254,7 @@ export const filtersToQueryNode = (filters: Partial<FilterType>): InsightQueryNo
             aggregation_axis_format: filters.aggregation_axis_format,
             aggregation_axis_prefix: filters.aggregation_axis_prefix,
             aggregation_axis_postfix: filters.aggregation_axis_postfix,
+            decimal_places: filters.decimal_places,
```
[the remark this answers is not in the document]
Comment (Twixes) [reply frontend/src/queries/nodes/InsightQuery/utils/filtersToQueryNode.ts:257]: Thanks, updated this too!

**Review by thmsobrmlr — APPROVED** [review]

**Comment by Twixes** [comment]
Thanks for taking a look @thmsobrmlr. All conversion and tests should be updated now.
Also I added one small quality of life improvement: "Decimal places" should now only show up in Trends insights that _might_ have fractional values (so e.g. "average property values", but not "total count").

**Comment by thmsobrmlr** [comment]
> Thanks for taking a look @thmsobrmlr. All conversion and tests should be updated now. Also I added one small quality of life improvement: "Decimal places" should now only show up in Trends insights that _might_ have fractional values (so e.g. "average property values", but not "total count").

Makes sense and looks good to me!

**Pull request merged**